utils/v2tofhirr4/mappings.bal

```diff
@@ -155,123 +151,145 @@ public isolated function segmentToFhir(string segmentName, hl7:Segment segment,
-    'source: hdToMessageHeaderSource(msh.msh3),
-    destination: [hdToMessageHeaderDestination(msh.msh5)],
-    eventCoding: (msh.msh9 is hl7v23:CM_MSG) ? msgToCoding(<hl7v23:CM_MSG>msh.msh9) : {},
-    language: (msh.msh19 is hl7v26:CWE|hl7v27:MSH|hl7v28:MSH) ? cweToCode(<hl7v26:CWE>msh.msh19) : ceToCode(<hl7v23:CE|hl7v231:CE|hl7v24:CE|hl7v25:CE|hl7v251:CE>msh.msh19),
-    eventUri: ""
+public isolated function mshToMessageHeader(hl7v2commons:Msh msh) returns international401:MessageHeader {
+    international401:MessageHeader messageHeader = {
+        'source: hdToMessageHeaderSource(msh.msh3),
+        destination: [hdToMessageHeaderDestination(msh.msh5)],
+        eventCoding: (msh.msh9 is hl7v23:CM_MSG) ? msgToCoding(<hl7v23:CM_MSG>msh.msh9) : {},
+        language: (msh.msh19 is hl7v26:CWE|hl7v27:MSH|hl7v28:MSH) ? cweToCode(<hl7v26:CWE>msh.msh19) : ceToCode(<hl7v23:CE|hl7v231:CE|hl7v24:CE|hl7v25:CE|hl7v251:CE>msh.msh19),
+        eventUri: ""
+    };
+    return messageHeader;
 };
 
 // --- Patient Administation
 public isolated function al1ToAllerygyIntolerance(hl7v2commons:Al1 al1) returns international401:AllergyIntolerance {
-    r4:Coding[] coding = [
-        {
-            code: al1.al11.toString(),
-            system: al1.al11.toString()
-        }
-    ];
 
-    international401:AllergyIntoleranceReaction[] allergyIntoleranceReaction = [];
-    string onset = "";
-    onset = al1.al16;
+    international401:AllergyIntoleranceReaction[] allergyIntoleranceReactions = [];
     if al1.al15 is hl7v23:ST {
-        allergyIntoleranceReaction = [
+
+        allergyIntoleranceReactions = [
             {
-                manifestation: [
-                    {
-                        text: <hl7v23:ST>al1.al15
-                    }
-                ],
-                onset: onset
+                manifestation: (al1.al15 != "") ? [
+                        {
+                            text: <hl7v23:ST>al1.al15
+                        }
+                    ] : [],
+                onset: (al1.al16 != "") ? al1.al16 : ()
             }
         ];
     } else if al1.al15 is hl7v231:ST[] {
         foreach hl7v231:ST reaction in <hl7v231:ST[]>al1.al15 {
-            allergyIntoleranceReaction.push({
-                manifestation: [
-                    {
-                        text: reaction
-                    }
-                ],
-                onset: onset
-            });
+            international401:AllergyIntoleranceReaction allergyIntoleranceReaction = {
+                manifestation: (reaction != "") ? [
+                        {
+                            text: reaction
+                        }
+                    ] : [],
+                onset: (al1.al16 != "") ? al1.al16 : ()
+            };
+            if allergyIntoleranceReaction.manifestation != [] || allergyIntoleranceReaction.onset != () {
+                allergyIntoleranceReactions.push(allergyIntoleranceReaction);
+            }
```

Comment:
Instead of creating the allergyIntoleranceReaction record and then doing checks on that, can we first check reaction and then create the record?